delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/jsonld/JsonLdAssembly.scala

```diff
@@ -0,0 +1,78 @@
+package ch.epfl.bluebrain.nexus.delta.sdk.jsonld
+
+import cats.Eq
+import cats.effect.IO
+import cats.syntax.all._
+import ch.epfl.bluebrain.nexus.delta.rdf.IriOrBNode.Iri
+import ch.epfl.bluebrain.nexus.delta.rdf.RdfError
+import ch.epfl.bluebrain.nexus.delta.sdk.jsonld.JsonLdRejection._
+import ch.epfl.bluebrain.nexus.delta.rdf.graph.Graph
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.api.JsonLdApi
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.context.{ContextValue, RemoteContext, RemoteContextResolution}
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.{CompactedJsonLd, ExpandedJsonLd}
+import ch.epfl.bluebrain.nexus.delta.sdk.model.jsonld.RemoteContextRef
+import io.circe.Json
+
+/**
+  * Result of the processing of the source from the [[JsonLdSourceProcessor]] which validates that the different
+  * representations and the id is valid or generated
+  * @param id
+  *   the identifier of the resource
+  * @param source
+  *   the original payload
+  * @param compacted
+  *   its compacted json-ld representation
+  * @param expanded
+  *   its expanded json-ld representation
+  * @param graph
+  *   its graph representation
+  * @param remoteContexts
+  *   it
+  */
+final case class JsonLdAssembly(
+    id: Iri,
+    source: Json,
+    compacted: CompactedJsonLd,
+    expanded: ExpandedJsonLd,
+    graph: Graph,
+    remoteContexts: Set[RemoteContextRef]
+) {
+
+  /**
+    * The collection of known types
+    */
+  def types: Set[Iri] = expanded.getTypes.getOrElse(Set.empty)
+}
+
+object JsonLdAssembly {
+
+  /**
+    * Defines the equality between two instances
+    *
+    *   - If the remote contexts and the local context are the same, then the compacted form will be the same
+    *   - If the graph forms are isomorphic then, the expanded form will be the same
+    */
+  implicit val jsonLdAssemblyEq: Eq[JsonLdAssembly] = Eq.instance { (jsonld1, jsonld2) =>
+    jsonld1.id == jsonld2.id &&
+    jsonld1.remoteContexts == jsonld2.remoteContexts &&
+    jsonld1.compacted.ctx == jsonld2.compacted.ctx &&
+    jsonld1.graph.isIsomorphic(jsonld2.graph) &&
+    jsonld1.source == jsonld2.source
+  }
```

Comment:
Why do we need to check the `id` here? Doesn't the graph isomorphism guarantee that already?

The following passes:

```scala
    "not be isomorphic when the id is different" in {
      val graph2 = graph.replaceRootNode(iri"http://other.id")
      graph.isIsomorphic(graph2) shouldEqual false
    }
```

Reply:
It is a fail fast thing, I tried to put the least expensive checks first

Reply:
Ok make sense. I wanted to check if there wasn't something I was misunderstanding about `isIsomorphic`